code/API_definitions/BYON-Webrtc-Events.yaml

```
@@ -0,0 +1,1456 @@
+          $ref: "#/components/schemas/SinkCredential"
+        types:
+          description: |-
+            Camara Event types eligible to be delivered by this subscription.
+            Note: for the Commonalities meta-release v0.4 we enforce to have only event type per subscription
+            then for following meta-release use of array MUST be decided at API project level.
+          type: array
+          minItems: 1
+          maxItems: 1
+          items:
+            $ref: "#/components/schemas/SubscriptionEventTypeNotification"
+        config:
+          $ref: "#/components/schemas/Config"
+      discriminator:
+        propertyName: protocol
+        mapping:
+          HTTP: "#/components/schemas/HTTPSubscriptionRequest"
+
+    Config:
+      description: |-
+        Implementation-specific configuration parameters needed by the subscription manager for acquiring
+        events. `subscriptionExpireTime`, `subscriptionMaxEvents`, `initialEvent` are predefined for CAMARA
+        specifications.
+
+        WebRTC specific event type attributes are included into `WebrtcSubscriptionDetail` schema.
+
+        Note: if a request is performed for several event type, all subscribed event will use same `config`
+        parameters.
+      type: object
+      required:
+        - subscriptionDetail
+      properties:
+        subscriptionDetail:
+          $ref: "#/components/schemas/WebrtcSubscriptionDetail"
+        subscriptionExpireTime:
+          type: string
+          format: date-time
+          example: 2023-01-17T13:18:23.682Z
+          description: The subscription expiration time (in date-time format) requested by the API consumer. Up to API project decision to keep it.
+        subscriptionMaxEvents:
+          type: integer
+          description: Identifies the maximum number of event reports to be generated (>=1) requested by the API consumer - Once this number is reached, the subscription ends. Up to API project decision to keep it.
+          minimum: 1
+          example: 5
+        initialEvent:
+          type: boolean
+          description: |-
+            Set to `true` by API consumer if consumer wants to get an event as soon as the subscription is created and current situation reflects event request.
+            Example: Consumer request Roaming event. If consumer sets initialEvent to true and device is in roaming situation, an event is triggered
+            Up to API project decision to keep it.
+
+    Protocol:
+      type: string
+      enum: ["HTTP"]
+      description: Identifier of a delivery protocol. Only HTTP is allowed for now
+      example: "HTTP"
+
+    SinkCredential:
+      description: A sink credential provides authentication or authorization information necessary to enable delivery of events to a target.
+      type: object
+      properties:
+        credentialType:
+          type: string
+          enum:
+            - PLAIN
+            - ACCESSTOKEN
+            - REFRESHTOKEN
```

https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#12-subscription-notification--event

Commonalities clearly defines "Type of the credential - MUST be set to ACCESSTOKEN.". 
Hence, to avoid any confusion, it is suggested to remove PlainCredential and RefreshTokenCredential from enum. 

```suggestion
          enum:
            - ACCESSTOKEN
```


**Consequence if not changed accordingly**:
Commonalities clearly defines "Type of the credential - MUST be set to ACCESSTOKEN.". This inconsistency can confuse developers, who may incorrectly assume that PLAIN and REFRESHTOKEN are valid options and attempt to use them. In fact, as they are defined as part of schema and the API documentation is generated from the schema, it will include all the options (ACCESSTOKEN, PLAIN, REFRESHTOKEN) which will create unnecessary ambiguity. Although the schema defines corresponding error such as GENERIC_400_INVALID_CREDENTIAL, having unnecessary schema definition will create unnecessary validation (ensures that only ACCESSTOKEN can be passed, and other shall be rejected) at GW.

---

Good one! thanks a lot!

---

Fixed at [93a4bcf](https://github.com/camaraproject/WebRTC/pull/53/commits/93a4bcf6f12c8fa4792f9be936cffe28c7e06d5b)